to-sovereign/tests/to_sovereign_unit_tests.rs

```diff
@@ -758,3 +893,126 @@ fn register_token_nonfungible_token() {
+        0,
+        EsdtTokenData::default(),
+    );
+
+    let operation_data = OperationData::new(1, OWNER_ADDRESS.to_managed_address(), None);
+
+    let operation = Operation::new(
+        TESTING_SC_ADDRESS.to_managed_address(),
+        vec![payment].into(),
+        operation_data,
+    );
+
+    let hash_of_hashes = state.get_operation_hash(&operation);
+
+    state.deploy_header_verifier(CHAIN_CONFIG_ADDRESS);
+    state.set_header_verifier_address(HEADER_VERIFIER_ADDRESS);
+
+    state.execute_operation(
+        hash_of_hashes,
+        operation,
+        Some("There is no registered ESDT address"),
+    );
+}
+
+#[test]
+fn execute_operation_setup_phase_not_completed() {
+    let mut state = ToSovereignTestState::new();
+    let config = EsdtSafeConfig::default_config();
+    state.deploy_contract(config);
+
+    let token_data = EsdtTokenData {
+        amount: BigUint::from(100u64),
+        ..Default::default()
+    };
+
+    let payment = OperationEsdtPayment::new(TokenIdentifier::from(TEST_TOKEN_ONE), 0, token_data);
+
+    let gas_limit = 1;
+    let function = ManagedBuffer::<StaticApi>::from("hello");
+    let args =
+        ManagedVec::<StaticApi, ManagedBuffer<StaticApi>>::from(vec![ManagedBuffer::from("1")]);
+
+    let transfer_data = TransferData::new(gas_limit, function, args);
+
+    let operation_data =
+        OperationData::new(1, OWNER_ADDRESS.to_managed_address(), Some(transfer_data));
+
+    let operation = Operation::new(
+        TESTING_SC_ADDRESS.to_managed_address(),
+        vec![payment].into(),
+        operation_data,
+    );
+
+    let operation_hash = state.get_operation_hash(&operation);
+    let hash_of_hashes = ManagedBuffer::new_from_bytes(&sha256(&operation_hash.to_vec()));
+
+    state.deploy_header_verifier(CHAIN_CONFIG_ADDRESS);
+    state.deploy_testing_sc();
+    state.set_header_verifier_address(HEADER_VERIFIER_ADDRESS);
+    state.set_esdt_safe_address_in_header_verifier(CONTRACT_ADDRESS);
+
+    let operations_hashes = MultiValueEncoded::from(ManagedVec::from(vec![operation_hash.clone()]));
+
+    state.deploy_chain_config(SovereignConfig::default_config());
+    state.change_validator_set(vec![ManagedBuffer::from("bls_1")]);
+    state.complete_setup_phase();
+    state.register_operation(ManagedBuffer::new(), &hash_of_hashes, operations_hashes);
+    state.execute_operation(hash_of_hashes, operation.clone(), None);
+
+    state
+        .world
+        .query()
+        .to(HEADER_VERIFIER_ADDRESS)
+        .whitebox(header_verifier::contract_obj, |sc| {
+            let operation_hash_whitebox = ManagedBuffer::new_from_bytes(&operation_hash.to_vec());
+            let hash_of_hashes =
+                ManagedBuffer::new_from_bytes(&sha256(&operation_hash_whitebox.to_vec()));
+
+            assert!(
+                sc.operation_hash_status(&hash_of_hashes, &operation_hash_whitebox)
+                    .get()
+                    == OperationHashStatus::NotLocked
```

Comment:
Please use `.is_empty()` here.
If you want to check if status is `NotLocked` you can check after `register_operation` and before `execute_operation`